pymemcache/client/base.py

```diff
@@ -237,15 +237,19 @@ def __init__(self,
         self.key_prefix = key_prefix
         self.default_noreply = default_noreply
         self.allow_unicode_keys = allow_unicode_keys
+        self._use_unix_socket = isinstance(server, str)
```

Comment:
Perhaps we could set `self.socket_protocol` here in the initializer instead, and then use that directly in `_connect()` below. The `self._use_unix_socket` flag isn't otherwise used.

Alternatively, we could inspect the type of `self.server` in `_connect()` and only set `protocol` locally within that function.